src/nvim/msgpack_rpc/channel.c

```diff
@@ -313,27 +312,39 @@ static void handle_request(Channel *channel, msgpack_object *request)
     return;
   }
 
-  // Retrieve the request handler
   MsgpackRpcRequestHandler handler;
-  Array args = ARRAY_DICT_INIT;
   msgpack_object *method = msgpack_rpc_method(request);
+  handler = msgpack_rpc_get_handler_for(method->via.bin.ptr,
+                                        method->via.bin.size);
 
-  if (method) {
-    handler = msgpack_rpc_get_handler_for(method->via.bin.ptr,
-                                          method->via.bin.size);
-    if (handler.fn == msgpack_rpc_handle_missing_method) {
-      String m = method->via.bin.size > 0
-        ? cbuf_to_string(method->via.bin.ptr, method->via.bin.size)
-        : cstr_to_string("<empty>");
-      ADD(args, STRING_OBJ(m));
-      handler.async = true;
-    } else if (!msgpack_rpc_to_array(msgpack_rpc_args(request), &args)) {
-      handler.fn = msgpack_rpc_handle_invalid_arguments;
-      handler.async = true;
-    }
-  } else {
-    handler.fn = msgpack_rpc_handle_missing_method;
-    handler.async = true;
+  // check method arguments
```

Comment:
is there a good reason this block was moved in front? The same ` !ERROR_SET(&error)` guard could be used with the original order right?

Reply:
No particular reason! I'll restore the original order